### PR TITLE
Improvement/error handling task creation

### LINKED ISSUE #282: Improve error, if ECS task can not be submitted
Improve Error trace, if the ECS worker can not create ECS tasks. One example would be if the Fargate vCpu quota is met.

# Traceback / Example

image

image

image

Currently an `IndexError` is returned in the stack. And in the Cloud UI State: Message is: `Submission failed. IndexError: list index out of range`

# Expectation / Proposal

More pass more details to the error. Maybe wrap the error in another error like `SubmissionError` 

- [x] I would like to [help contribute](https://PrefectHQ.github.io/prefect-aws/#contributing) a pull request to resolve this!

### PR DESCRIPTION
**Pull request by jeanluciano**
Introduces logic to catch failures from the run task response. It raises a `RuntimeError` when it finds the list of failures populated of if either `tasks` or `failures` which will help with the index errors that can come up.

Closes #282

### Checklist

- [ x ] References any related issue by including "Closes #<Issue Number>" or "Closes <Issue URL>".
  - If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect-aws/issues/new/choose) first.
- [ x ] Includes tests or only affects documentation.
- [ x ] Passes `pre-commit` checks.
  - Run `pre-commit install && pre-commit run --all` locally for formatting and linting.
- [ ] Includes screenshots of documentation updates.
  - Run `mkdocs serve` view documentation locally.

### CODE REVIEW

**Review by desertaxle — CHANGES_REQUESTED** [review]

File: tests/workers/test_ecs_worker.py [hunk 1]
```diff
@@ -2316,3 +2317,27 @@ async def test_mask_sensitive_env_values():
...
+
+    async with ECSWorker(work_pool_name=work_pool_name) as worker:
+        result = await run_then_stop_task(worker, configuration, flow_run)
+
+    assert result.status_code == 0
+    _, task_arn = parse_identifier(result.identifier)
+
+    task = describe_task(ecs_client, task_arn)
+    task_definition = describe_task_definition(ecs_client, task)
+    assert task_definition["family"] == family
```
[the remark this answers is not in the document]
Comment (jeanluciano) [reply tests/workers/test_ecs_worker.py:2343]: This test is from me checking out main as I didn't write any test for this.

**Review by desertaxle — APPROVED** [review]
LGTM!

**Comment by MohammedSiddiqui** [comment]
@jeanluciano thanks for taking care of this!
@desertaxle any idea when these changes can be available now in a new release?

**Comment by rmnvncnt** [comment]
Thanks a lot for that one!

**Pull request merged**